### PR TITLE
Arm osx bootstrap

### PR DESCRIPTION
**Pull request by danielhollas**
Trying to see if the bootstrap build works for ARM64 on OSX as well. Build on top of #158 

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/build.sh [hunk 1]
```diff
@@ -9,20 +13,67 @@ if [[ "${target_platform}" == osx-* ]]; then
...
+  # the bootstrap system needs to be compiled with the guest (x86) toolchain.
+  # Otherwise, configure fails with:
+  # error: Cannot both cross compile and build a bootstrap system
+  local CC CXX CPP LD AR RANLIB
+  CC=${CC_FOR_BUILD}
+  CXX=${CXX_FOR_BUILD}
+  LD=${CC_FOR_BUILD//gnu-cc/gnu-ld}
+  CPP=${CC_FOR_BUILD//gnu-cc/gnu-cpp}
+  AR=${CC_FOR_BUILD//gnu-cc/gnu-ar}
+  RANLIB=${CC_FOR_BUILD//gnu-cc/gnu-ranlib}
```
Comment (jakirkham) [inline recipe/build.sh:28]: These may be different for macOS as we use a Clang-based toolchain there

Comment (danielhollas) [reply recipe/build.sh:28]: Indeed, I am just looking at the failed build. :sweat_smile: 
clang also doesn't like empty CFLAGS. I'll try to fix these tomorrow.

Comment (danielhollas) [reply recipe/build.sh]: The build now works! :tada: But it seems we'll have to skip tests in this case.

**Comment by danielhollas** [comment]
@jakirkham I wanted to manually test the OSX arm64 build from this PR locally on a M2 Mac machine. I tried to follow the [Downloading prebuilt packages from CI](https://conda-forge.org/docs/maintainer/updating_pkgs/#downloading-prebuilt-packages-from-ci) section of the Docs.
I found it quite confusing and lacking enough details so perhaps I am doing something wrong. 
In the last commit, I've added the following to `conda-forge.yml` to enable artifact upload on Azure pipelines

```
azure:
   store_build_artifacts: true
```

But I still don't see any uploaded artifacts. Would you mind taking a look?

**Comment by danielhollas** [comment]
@conda-forge-admin please rerender

**Comment by jakirkham** [comment]
There are some screenshots with explanations in this issue ( https://github.com/conda-forge/conda-forge.github.io/issues/1424 ), which can help in downloading and testing these

**Comment by jakirkham** [comment]
Should add if you have recommendations on how to make the docs clearer we would welcome them  🙂

(even more so if in PR form 😉)

**Comment by danielhollas** [comment]
Haha, yes I was thinking of submitting a PR as I found the instructions very confusing. :-) I actually found the issue and the screenshots definitely helped. Do you think they should be in some form be included in the docs?

Ultimately the step I was missing is the rerender, which I think is not explicitly mentioned.

**Comment by jakirkham** [comment]
Yeah that was certainly one idea we had for the docs. If you find them helpful, agree we should add them 🙂

Calling out re-rendering in the docs is also a good idea 👍

**Comment by danielhollas** [comment]
Another question I had about this, is the artifact upload available only on Azure pipelines? I haven't seen the option for the other CI providers.

**Comment by jakirkham** [comment]
Yeah think it is just GitHub Actions and Azure

It's probably possible to implement on other CI providers, but am not aware of current work on that

**Comment by danielhollas** [comment]
Now that I've been able to download the artifact I could install the package locally on Mac M2. Unfortunately I am getting this error:

```
$ erl --version
dyld[36883]: Library not loaded: @rpath/libtinfo.6.dylib

  Referenced from: <64D3CFF0-45A6-37FA-AE4C-05C25E3CF527> miniconda3/envs/erlang-osx/lib/erlang/erts-15.0/bin/beam.smp

  Reason: tried: '/Users/veronikajuraskova/miniconda3/envs/erlang-osx/lib/erlang/erts-15.0/bin/../../../libtinfo.6.dylib' (no such file),
'/Users/veronikajuraskova/miniconda3/envs/erlang-osx/lib/erlang/erts-15.0/bin/../../../libtinfo.6.dylib' (no such file), '/usr/local/lib/libtinfo.6.dylib' (no such file), 
'/usr/lib/libtinfo.6.dylib' (no such file, not in dyld cache)
```

Googling around a bit it seems that libtinfo should come from ncurses library. Suspiciously, listing installed packages with `mamba list` only shows `erlang` and not other dependencies. So I suspect that runtime dependencies are somehow not correctly specified for the package. I'll need to have a closer look.

**Comment by danielhollas** [comment]
Okay, it seems that conda doesn't install package dependencies when installing from `*.conda` package file?? Installing `ncurses` and `zlib` manually solved the issue and we verified that `erl --version` works. 

Unfortunately, the artifact doesn't contain the compiled test suite so we haven't run it. I don't know how much work it would be to try to upload the test suite as artifact as well, and if it is worth the hassle. 

Apart from that I think this is ready to go. :rocket:

**Comment by jakirkham** [comment]
That's right. Would use `file://<path to build_artifacts` as the channel to install `erlang`

This should pull in dependencies

**Comment by jakirkham** [comment]
@conda-forge-admin , please re-render

File: conda-forge.yml [hunk 2]
```diff
@@ -1,10 +1,15 @@
...
 github:
   branch_name: main
   tooling_branch_name: main
-conda_build:
-  pkg_format: '2'
+os_version:
+  linux_64: cos7
+test: native_and_emulated
+azure:
+  store_build_artifacts: true
```
Comment (jakirkham) [inline conda-forge.yml:15]: Note: Turning artifact storage back off now that testing is complete

```suggestion
```

**Comment by jakirkham** [comment]
@conda-forge-admin , please re-render

**Comment by jakirkham** [comment]
@conda-forge-admin , please re-render

**Comment by jakirkham** [comment]
@conda-forge-admin , please re-render

File: recipe/build.sh [hunk 3]
```diff
@@ -1,4 +1,8 @@
 #!/bin/bash
+# Get an updated config.sub and config.guess
+cp $BUILD_PREFIX/share/gnuconfig/config.* ./erts/autoconf
+cp $BUILD_PREFIX/share/gnuconfig/config.* ./make/autoconf
+cp $BUILD_PREFIX/share/gnuconfig/config.* ./lib/common_test/test_server
 set -ex
```
Comment (jakirkham) [inline recipe/build.sh:6]: Let's do `set` first

```suggestion
set -ex

# Get an updated config.sub and config.guess
cp $BUILD_PREFIX/share/gnuconfig/config.* ./erts/autoconf
cp $BUILD_PREFIX/share/gnuconfig/config.* ./make/autoconf
cp $BUILD_PREFIX/share/gnuconfig/config.* ./lib/common_test/test_server
```

**Comment by jakirkham** [comment]
@conda-forge-admin , please re-render

**Comment by jakirkham** [comment]
Thanks Daniel! 🙏

Have merged in the changes from PR ( https://github.com/conda-forge/erlang-feedstock/pull/159 ) and resolved conflicts

**Pull request merged**